src/v/config/configuration.cc

```diff
@@ -118,6 +118,16 @@ configuration::configuration()
        .example = "31536000000",
        .visibility = visibility::tunable},
       std::nullopt)
+  , log_storage_max_size_high_watermark(
```

Comment:
Naming: usually "high watermark" means the highest value something has ever reached.  I may be taking this very literally because I went to university in a place that floods regularly and the pubs have a literal high watermark on the wall.

Maybe something like "log_storage_trim_threshold" or something like that?  It's not particularly specific but makes it more obvious that it's a threshold at which something triggers.

Reply:
(we have the same naming quirk in tiered storage cache where `_cache_size_low_watermark` is not a low watermark but a percentage to which space gets trimmed)

Reply:
sgtm. i had it as low-watermark originally, but changed it to high-watermark. neither seemed great.